File: tvos/plugin/ios/BitmovinRNPlayer/PlayerBridge.m
//
//  PlayerBridge.m
//  LightApp
//
//  Created by Afanasiev, Anatolii on 21/01/2020.
//  Copyright © 2020 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(PlayerBridge, RCTViewManager)

// callback to reactNative
RCT_EXPORT_VIEW_PROPERTY(showSettingsEvent, RCTDirectEventBlock)

// TEST!
RCT_EXTERN_METHOD(playableItem:(NSDictionary)dictionary)
RCT_EXTERN_METHOD(onKeyChanged:(NSDictionary)dictionary)
RCT_EXTERN_METHOD(pluginConfiguration:(NSDictionary)dictionary)
RCT_EXTERN_METHOD(onSettingSelected:(NSDictionary)dictionary)
@end

File: tvos/plugin/ios/BitmovinRNPlayer/PlayerBridge.swift
//
//  PlayerBridge.swift
//  LightApp
//
//  Created by Afanasiev, Anatolii on 21/01/2020.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation
import BitmovinPlayer

@objc(PlayerBridge)
class PlayerBridge: RCTViewManager {
  
  let bulbView: PlayerView = PlayerView()

  override func view() -> UIView! {
    return self.bulbView
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc
  func playableItem(_ dictionary: NSDictionary) {
    self.bulbView.playableItem = dictionary
  }

  @objc
  func onKeyChanged(_ dictionary: NSDictionary) {
    self.bulbView.onKeyChanged = dictionary
  }

  @objc
  func pluginConfiguration(_ dictionary: NSDictionary) {
    self.bulbView.pluginConfiguration = dictionary
  }

  @objc
  func onSettingSelected(_ dictionary: NSDictionary) {
    self.bulbView.onSettingSelected = dictionary
  }
}
